### ISSUE TITLE
jar confict  in spring-boot-3.0.0-m3 

### ISSUE THREAD

**Issue by saturnLong** [spring-projects/spring-boot#31655]
the version definition in spring-boot-m3 bom：
reactor 3.5.0-M2  confict whith  lettuce  6.1.8.RELEASE

java.lang.NoSuchMethodError: 'reactor.core.publisher.Mono reactor.core.publisher.Mono.subscriberContext()'
	at io.lettuce.core.tracing.Tracing.getContext(Tracing.java:84) ~[lettuce-core-6.1.8.RELEASE.jar:6.1.8.RELEASE]

**Comment by wilkinsona** [comment]
Thanks for the report. This will have to be addressed in Lettuce (cc @mp911de) and Reactor (cc @simonbasle). We'll pick up any resulting Lettuce or Reactor releases in due course.

**Comment by simonbasle** [comment]
yes, Lettuce 6.1.x apparently depends on reactor-core 3.3.x (which is EoL) and causes this issue. From what I can see, in the current `main` @mp911de is working towards a Lettuce 6.2.x which depends on reactor-core 3.4.x for now, which should eliminate this particular problem. I wonder if there's a plan to support reactor-core 3.5 directly instead, but the two lines should be compatible.

**Comment by mp911de** [comment]
For the time being, please configure `ClientResources.eventBus` with an [empty bus emitter](https://github.com/spring-projects/spring-data-redis/blob/862decfdbe693ff58c671aff69e265903d1273b4/src/test/java/org/springframework/data/redis/test/extension/LettuceTestClientResources.java#L39-L49) if you don't actively consume Lettuce client events. 

As Simon mentioned, Lettuce 6.2 is in the making and we cannot upgrade to Reactor 3.4 or 3.5 in a Lettuce 6.1.x minor release.

**Comment by wilkinsona** [comment]
@mp911de The error reported here doesn't appear to be related to the `EventBus`. I only see a problem with `EventBus` and its use of `DirectProcessor` when upgrading Reactor from `2022.0.0-M2`.
